provider/provider.go

```diff
@@ -79,3 +81,21 @@ func (s *SweepingProvider) StartProviding(force bool, keys ...mh.Multihash) {
 func (s *SweepingProvider) StopProviding(keys ...mh.Multihash) {
 	// TODO: implement me
 }
+
+// ProvideState encodes the current relationship between this node and `key`.
+type ProvideState uint8
+
+const (
+	StateUnknown  ProvideState = iota // we have no record of the key
+	StateQueued                       // key is queued to be provided
+	StateProvided                     // key was provided at least once
+)
+
+// ProvideStatus reports the provider’s view of a key.
+//
+// When `state == StateProvided`, `ts` is the wall‑clock time of the most recent
+// successful provide operation (UTC).
+// For `StateQueued` or `StateUnknown`, `ts` is the zero `time.Time`.
+func (s *SweepingProvider) ProvideStatus(key mh.Multihash) (state ProvideState, ts time.Time) {
```

Comment:
Should we consider adding an explicit interface definition?

```go
type ProvideStatusReporter interface {
  ProvideStatus(key mh.Multihash) (state ProvideState, ts time.Time)
}
```

Reply:
or... maybe make it part of `type DHTProvider interface` (https://github.com/libp2p/go-libp2p-kad-dht/pull/1093) to enforce implementations to track this information? Any reason we would not want it to be enforced at the level of that interface?

Reply:
> Any reason we would not want it to be enforced at the level of that interface?

The `boxo/provider` implementation doesn't (and I doubt we want to spend resources to) implement such a method, and we want a shared interface between `boxo/provider` and the new `SweepingProvider`.

The `DHTProvider` is expected to be a temporary interface until `boxo/provider` is deprecated. Then, we can use the `SweepingProvider` struct directly, no need to have an interface.